### PR TITLE
add app discovery feature

### PR DESCRIPTION
**Pull request by katzio**

### CODE REVIEW

File: kinappserver/models/__init__.py [hunk 1]
```diff
@@ -16,3 +16,4 @@
 from .blacklisted_phone_numbers import *
 from .system_config import *
 from .category import *
+from .app_descovery import *
```
Comment (v1thesource) [inline kinappserver/models/__init__.py]: "des"covery instead of "dis". Intentional?

Comment (katzio) [reply kinappserver/models/__init__.py]: oops

Comment (berryve) [inline kinappserver/models/__init__.py]: @katzio the file name is still app_descovery .... pls fix

Comment (katzio) [reply kinappserver/models/__init__.py]: wut, It's not

File: kinappserver/tests/app_discovery.py [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+        self.assertEqual(len(data[0]['apps']), 0)
+
+        # create an androdi and ios app
+        app = {
+            'skip_image_test': 'true',
+            'app_identifier': '1',
+            'title': 'app title',
+            'subtitle': 'subtitle',
+            'app_category_id': '0',
+            'os_type': 'android, iOS',
```
Comment (berryve) [inline kinappserver/tests/app_discovery.py]: how would that work ? we need an additional app_url in the metadata in such cases

File: kinappserver/views_private.py [hunk 3]
```diff
@@ -884,3 +892,57 @@ def migrate_tasks_to_task20():
     task20_migrate_tasks()
     return jsonify(status='ok')
 
+
+@app.route('/app_discovery/add_discovery_app_category', methods=['POST'])
```
Comment (berryve) [inline kinappserver/views_private.py]: could simplify the endpoint to: '/app_discovery/add_category'

File: kinappserver/views_private.py [hunk 4]
```diff
@@ -884,3 +892,57 @@ def migrate_tasks_to_task20():
...
+    except Exception as e:
+        print('exception: %s' % e)
+        raise InvalidUsage('bad-request')
+    if set_discovery_app_active(app_id, set_active):
+        return jsonify(status='ok')
+    else:
+        raise InvalidUsage('failed to activate discovery app')
+
+
+@app.route('/app_discovery/add_discovery_app', methods=['POST'])
```
Comment (berryve) [inline kinappserver/views_private.py]: could simplify the endpoint to /app_discovery/add_app

**Pull request merged**